### DISCUSSION TITLE
How to edit forms with checkboxes ?

### DISCUSSION THREAD

**Discussion by ThierryM** [sqlpage/SQLPage#339] [Q&A] [answered]
Hi,
I'm trying to edit existing data on a boolean field with checkboxes in sqlite database. But when I load the form, the checkboxes are always unchecked (as if the value was 0).
When I create a new line, there's no problem : if I check the checkbox, 1 is saved in database and if it's unchecked the saved value is NULL.
So, in order to edit an existing record, how can I load a form with checkboxes checked if the initial value is 1 in the field loaded ?
Thanks for your help. Regards

**Comment by lovasoa** [comment]
Hello! You can use the "checked" row-level attribute of the form component.

**Comment by DSMejantel** [comment, marked as answer]
Helloc !

Like this in ruche.sql : 

```sql
    SELECT 
'Mort de la colonie' AS label, 
'mort' AS name, 
CASE WHEN disparition=1 THEN TRUE ELSE '' END as checked, 
'checkbox' as type, 
1 as value, 
3 as width 
from colonie where numero = $id
```

**Reply by lovasoa** [reply]
@DSMejantel wouldn't it work with just `disparition=1 as checked` instead of `CASE WHEN disparition=1 THEN TRUE ELSE '' END as checked` ?

**Comment by DSMejantel** [comment]
Hi !
You're right. It is shorter and it works

**Comment by ThierryM** [comment]
Hi,
Thanks a lot for your explanations !
Now it's working.
Regards.
